app/src/main/java/org/thoughtcrime/securesms/notifications/DefaultMessageNotifier.kt

```diff
@@ -285,12 +281,17 @@ class DefaultMessageNotifier : MessageNotifier {
 
         builder.putStringExtra(LATEST_MESSAGE_ID_TAG, messageIdTag)
 
-        val text = notifications[0].text
+        val notificationText = notifications[0].text
+
+        // For some reason, even when we're starting a call we get a missed call notification - so we'll bail before that happens.
+        // TODO: Probably better to fix this at the source so that this never gets called rather then here - do this.
+        val missedCallString = Phrase.from(context, R.string.callsMissedCallFrom).put(NAME_KEY, notifications[0].recipient.name).format()
+        if (ApplicationContext.isAppVisible && notificationText == missedCallString) { return }
```

Comment:
It seems a bit fragile to compare a user facing text to determine the business logic?